HTML/nested.html

```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Nested Classes Example</title>
+</head>
+<body>
+    <h1>Nested Classes Example</h1>
+    <div id="output"></div>
+
+    <script>
+        class OuterClass {
+            constructor(name) {
+                this.name = name;
+            }
+
+            greet() {
+                console.log(`Hello from ${this.name}`);
+            }
+
+            class InnerClass {
+                constructor(innerName) {
+                    this.innerName = innerName;
+                }
+
+                innerGreet() {
+                    console.log(`Hello from inner class, ${this.innerName}`);
+                }
+            }
+        }
+
+        const outer = new OuterClass('Outer');
+        outer.greet();
+
+        const inner = new OuterClass.InnerClass('Inner');
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Potential issue with InnerClass instantiation</b></div>
<div id="fix">

The instantiation of `InnerClass` might not work as expected. Consider modifying how `InnerClass` is accessed.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -22,7 +22,7 @@
          }
 
          class InnerClass {
 -            constructor(innerName) {
 +            constructor(outerInstance, innerName) {
                  this.innerName = innerName;
              }
 
 @@ -33,7 +33,7 @@
          const outer = new OuterClass('Outer');
          outer.greet();
 
 -        const inner = new OuterClass.InnerClass('Inner');
 +        const inner = new outer.InnerClass(outer, 'Inner');
          inner.innerGreet();
 
          document.getElementById('output').innerText = 'Check the console for greetings from ${outer.name} and ${inner.innerName}';
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/21#issuecomment-2437737202>#10c142</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged